### PR TITLE
[1] Use HL7 Validator from Inferno

### PR DESCRIPTION
**Pull request by dczulada**
Pull requests into Cypress require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [ ] This pull request describes why these changes were made.
- [ ] Internal ticket for this PR:
- [ ] Internal ticket links to this PR
- [ ] Code diff has been done and been reviewed
- [ ] Tests are included and test edge cases
- [ ] Tests have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: lib/validators/hl7_validator.rb [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      errors = []
+      warnings = []
+      information = []
+
+      issues.each do |iss|
+        if iss.severity == 'information' || iss.code == 'code-invalid' || ISSUE_DETAILS_FILTER.any? { |filter| filter.match?(iss&.details&.text) }
+          information << issue_message(iss)
+        elsif iss.severity == 'warning'
+          warnings << issue_message(iss)
+        else
```
Comment (lmd59) [inline lib/validators/hl7_validator.rb:66]: might be worth a specific `elsif iss.severity == 'error'` here in case a new issue severity is ever added.

File: lib/validators/hl7_validator.rb [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+      {
+        errors: errors,
+        warnings: warnings,
+        information: information
+      }
+    end
+
+    def issue_message(issue)
+      # location = if issue.respond_to?(:expression)
```
Comment (lmd59) [inline lib/validators/hl7_validator.rb:79]: do we want a TODO here or to ditch the commented code?

File: lib/validators/validation.rb [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+        elsif resource&.type&.coding&.any? { |coding| coding.code == 'SNF' }
+          return DEFINITIONS[BLUEBUTTON_URIS[:snf]]
+        end
+      end
+
+      # Otherwise, guess the first profile that matches on resource type
+      candidates.first
+    end
+
+    def self.guess_r4_profile(resource)
```
Comment (lmd59) [inline lib/validators/validation.rb:157]: I assume this profile matching stuff is copied in from somewhere? If so, can we put a comment sourcing where it's from? (might help with future maintainability)

**Review by lmd59 — APPROVED** [review]
